Update install.sh to work on other distros

Dalosuuu opened the pull request:
Cheers,

First, nice tool, I used it for one of the room you built in tryhackme and it's pleasant for automating part of the enum, but I had an issue when using install.sh in the Tryhackme's Attackbox since its Ubuntu and some of your install commands assume you are using kali and I had to do some manual install of packages. I made some edits to the install.sh so it works on other distros too.

changes and why

- added set -euo pipefail to make the script stop on errors or bad variables
- Fixed pip install logic: removed --break-system-packages since it caused issues in our Ubuntu environment due to unsupported pip version or restricted setup
- now checks if pip3 or pip exists, installs pip if not
- replaced repeated apt installs with a function (install_tool) to handle apt, snap, or git clone
- added fallback installs for seclists, impacket, and feroxbuster when apt doesn’t work
- changed hardcoded chown 1000:1000 to use the current user's uid/gid so it works for everyone
- updated the ctfenum launcher:
- moved logic into a heredoc for cleaner script
- changed "$1" to "$@" so it passes all arguments
- still puts it in /usr/sbin, but could be changed to /usr/local/bin later if needed
- kept everything else mostly the same, just safer and more compatible outside kali

josemlwdf reviewed (approved):
thanks for this

josemlwdf reviewed (changes requested):
- Enforce use of pip3 by removing this check:
elif command -v pip &>/dev/null; then
    PIP_CMD="pip"
	
- --break-system-packages i'm almost sure WSL asks for this one to work. Let's go without it for now as you suggested????

- Change impacket repo url to this one:
https://github.com/fortra/impacket

- Install feroxbuster with these methods:
sudo apt install -y feroxbuster

Dalosuuu commented:
Great observation didn't notice the redirect from SecureAuthCorp to Fortra. This is outdated [https://book.ice-wzl.xyz/domain-controllers/impacket-install](https://book.ice-wzl.xyz/domain-controllers/impacket-install) , I didn't noticed this when looking for installation alteratives.

I was testing the snapd forgot to change it, also that version of feroxbuster in snapd is old, changed to apt and it works now.

You are right about wsl error because is treating python packages as externally managed env if we don't include "--break-system-packages", we could leave it but it's just kinda dangerous for some distros from what I could find, we could check for wsl before using pip by checking /proc/version like this:

`grep -qi microsoft /proc/version`
Based on exit return status we can determine if we are on wsl or not

Also noticed we are checking for SecLists command but SecList is supposed to be a directory not a command so I changed that logic.

josemlwdf commented:
seclists is a command :)

![image](https://github.com/user-attachments/assets/f41014fb-c089-4431-919f-b5a0d358a529)

josemlwdf commented:
i have changed almost everything as you suggested. thanks for this

Dalosuuu commented:
> seclists is a command :)
> 
> ![image](https://private-user-images.githubusercontent.com/60181010/447807636-f41014fb-c089-4431-919f-b5a0d358a529.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgzNTY2MjcsIm5iZiI6MTc0ODM1NjMyNywicGF0aCI6Ii82MDE4MTAxMC80NDc4MDc2MzYtZjQxMDE0ZmItYzA4OS00NDMxLTkxOWYtYjVhMGQzNThhNTI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTI3VDE0MzIwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0Nzc5Nzk4NDM3ZjcyNmU2MDMyZDkwNDFhMjdlOTY0OWI4MWYyNmUxYzQ3M2U4OWFmNTE4OGE2MjhmYWQ1NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QCzGTnkYYGvY07unvbb0tgr9_1oMRY5IlFrtFKCwhFs)

First time I see that lol, didn't know they had a command. I think that's a kali thing tho [https://www.kali.org/tools/seclists/](https://www.kali.org/tools/seclists/) From what I see it is a tool descriptor from the kali-linux-tools system. When I was testing this it was not working on ubuntu, so I thought it was wrong because of that and since your python modules are using the path I went to check for the path too. 

Im happy to help, great work keep it up!

Dalosuuu commented:
You should definitely encourage(recommend) the use of your tools in your rooms on TryHackme, I think it will be a great way of exposing them.
